kyo-core/jvm/src/main/scala/kyo/Path.scala

```diff
@@ -331,6 +334,18 @@ class Path(val path: List[String]):
     def walk(maxDepth: Int)(using Frame): Stream[Path, IO] =
         Stream.init(IO(JFiles.walk(toJava).toScala(LazyList).map(path => Path(path.toString))))
 
+    override def hashCode(): Int =
+        val prime  = 31
+        var result = 1
+        result = prime * result + (if path == null then 0 else path.hashCode)
+        result
+    end hashCode
+
+    override def equals(obj: Any): Boolean = obj match
+        case that: Path =>
+            (this eq that) || (that.isInstanceOf[Path] && this.path == that.path)
```

Comment:
`that.isInstanceOf[Path]` should be unnecessary due to the match.

Why is the generated one not good enough?

Reply:
yep, it's unnecessary